### DISCUSSION TITLE
Is check_for_resize necessary if using poll_input?

### DISCUSSION THREAD

**Discussion by groves** [wezterm/wezterm#2514] [Q&A] [answered]
The docs on [check_for_resize](https://docs.rs/termwiz/latest/termwiz/terminal/buffered/struct.BufferedTerminal.html#method.check_for_resize) make me think I should call it periodically in my main loop. Looking in the Unix terminal code, I can see [handling of SIGWINCH in poll_input](https://github.com/wez/wezterm/blob/main/termwiz/src/terminal/unix.rs#L432). The latter is making me think I can ignore `check_for_resize` if I'm using `poll_input`. Is that correct?

I don't see any resize handling in the Windows terminal implementation. Does that mean I shouldn't expect to get resize events on Windows, or am I missing something? (not a problem if it's not currently implemented, just want to understand what to expect)

**Comment by wez** [comment, marked as answer]
Yes; while your application will receive resize events via `poll_input`, `BufferedTerminal` doesn't intercept those, and in any case, there is no guarantee that `poll_input` would even be used in a given application.

`check_for_resize` needs to get called in order for the surface maintained inside `BufferedTerminal` to be updated to reflect the size of the underlying terminal.

Regarding windows: yes, resize events are reported via `poll_input`; they are received and processed by [this code](https://github.com/wez/wezterm/blob/47f498768759750a0152fe6d76acb2a5b0ef7921/termwiz/src/input.rs#L1085)

**Reply by groves** [reply]
Makes sense. Thanks!
